chat-sdk-firebase-adapter/src/main/java/co/chatsdk/firebase/wrappers/ThreadWrapper.java

```diff
@@ -315,37 +315,38 @@ public void messagesOff() {
         FirebaseReferenceManager.shared().removeListeners(ref);
     }
 
-    public Observable<Thread> metaOn () {
-
+    public Observable<Thread> metaOn() {
         return Observable.create((ObservableOnSubscribe<Thread>) e -> {
-            DatabaseReference ref = FirebasePaths.threadMetaRef(model.getEntityID());
-            FirebaseReferenceManager.shared().addRef(ref, ref.addValueEventListener(new FirebaseEventListener().onValue((snapshot, hasValue) -> {
-                if(hasValue &&  snapshot.getValue() instanceof Map) {
-                    Map<String, Object> value = (Map<String, Object>) snapshot.getValue();
-                    for (String key : value.keySet()) {
-                        if (value.get(key) instanceof String) {
-                            model.setMetaValue(key, (String) value.get(key));
-                        }
-                    }
-                }
+
+            metaOff();
+
+            final DatabaseReference threadMetaRef = FirebasePaths.threadMetaRef(model.getEntityID());
+
+            if (FirebaseReferenceManager.shared().isOn(threadMetaRef)) {
                 e.onNext(model);
-            })));
+            }
+
+            ValueEventListener listener = threadMetaRef.addValueEventListener(new FirebaseEventListener().onValue((snapshot, hasValue) -> {
```

Comment:
This should be in the `else` statement. If meta is already on, we don't need to turn it on again.